geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedFeatureSourceView.scala

```diff
@@ -45,8 +45,14 @@ class MergedFeatureSourceView(
 
   override def getSchema: SimpleFeatureType = sft
 
-  override def getCount(query: Query): Int =
-    sources.map { case (source, filter) => source.getCount(mergeFilter(query, filter)) }.sum
+  override def getCount(query: Query): Int = {
+    val total = sources.map { case (source, filter) => source.getCount(mergeFilter(query, filter)) }.sum
```

Comment:
just thought of another issue... if some of these return -1 (indicating count is expensive), the result might be unexpected (i.e. -1 + 1 = 0). should we just return -1 in that case, or filter out the -1's and return the stores that return counts? or something else?

Reply:
Maybe something like
`val total = sources.map { case (source, filter) => source.getCount(mergeFilter(query, filter)) }.filter(_<0).sum` ?

Reply:
This is a good corner case to think through!  Good eye!

Reply:
since I think you're out, i made an executive decision to return -1 if any of the stores return -1, on the theory that `undefined + n = undefined`...